src/core/Database.h

```diff
@@ -99,76 +95,99 @@ class Database : public QObject
     bool containsDeletedObject(const DeletedObject& uuid) const;
     void setDeletedObjects(const QList<DeletedObject>& delObjs);
 
+    bool hasKey() const;
+    bool verifyKey(const QSharedPointer<CompositeKey>& key) const;
```

Comment:
It does verify the key. It does this in the Database class, because key and challengeResponseKey are two separate properties of the Database's Data struct. This method is prior art and I merely moved its declaration.